File: lib/widgets/reminder_bottom_sheet.dart
import 'package:flutter/material.dart';

enum _ReminderStep { main, autoSub }

class ReminderBottomSheet extends StatefulWidget {
  final ValueChanged<String>? onContinue;

  const ReminderBottomSheet({super.key, this.onContinue});

  @override
  State<ReminderBottomSheet> createState() => _ReminderBottomSheetState();
}

class _ReminderBottomSheetState extends State<ReminderBottomSheet> {
  _ReminderStep _step = _ReminderStep.main;
  int? _selected;
  int? _autoSubSelected;
  String? _mainSelection;

  void _selectAuto() {
    setState(() {
      _mainSelection = 'Auto Reminders';
      _step = _ReminderStep.autoSub;
    });
  }

  void _selectManual() {
    setState(() {
      _mainSelection = 'Manual Reminders';
      _selected = 1;
    });
  }

  void _back() {
    setState(() {
      _step = _ReminderStep.main;
      _autoSubSelected = null;
    });
  }

  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: EdgeInsets.only(
        left: 16,
        right: 16,
        top: 16,
        bottom: MediaQuery.of(context).viewInsets.bottom + 16,
      ),
      child: Column(
        mainAxisSize: MainAxisSize.min,
        children: [
          Row(
            mainAxisAlignment: MainAxisAlignment.spaceBetween,
            children: [
              if (_step == _ReminderStep.autoSub)
                IconButton(
                  icon: const Icon(Icons.arrow_back),
                  onPressed: _back,
                ),
              Text(
                _step == _ReminderStep.main
                    ? 'Select Type of Reminder'
                    : 'Auto Reminders',
                style: const TextStyle(fontWeight: FontWeight.bold, fontSize: 18),
              ),
              IconButton(
                icon: const Icon(Icons.close),
                onPressed: () => Navigator.of(context).pop(),
              ),
            ],
          ),
          const SizedBox(height: 12),
          AnimatedSwitcher(
            duration: const Duration(milliseconds: 200),
            child: _step == _ReminderStep.main
                ? _mainStep()
                : _autoSubStep(),
          ),
          const SizedBox(height: 8),
          Text(
            'Reminders will be sent via email and SMS based on your configured schedule.',
            style: TextStyle(fontSize: 12, color: Colors.grey[600]),
          ),
          const SizedBox(height: 16),
          SizedBox(
            width: double.infinity,
            height: 48,
            child: FilledButton(
              onPressed: _canContinue() ? () {
                final type = _mainSelection == 'Manual Reminders' ? 'manual' : 'auto';
                Navigator.of(context).pop();
                widget.onContinue?.call(type);
              } : null,
              child: const Text('CONTINUE'),
            ),
          ),
        ],
      ),
    );
  }

  bool _canContinue() {
    if (_mainSelection == 'Auto Reminders') return _autoSubSelected != null;
    return _selected == 1;
  }

  Widget _mainStep() {
    return Column(
      children: [
        _optionCard(
          icon: Icons.autorenew,
          title: 'Auto Reminders',
          description: 'Automatically send reminders based on predefined rules and schedules.',
          isSelected: _mainSelection == 'Auto Reminders',
          onTap: _selectAuto,
          showBadge: true,
        ),
        const SizedBox(height: 8),
        _optionCard(
          icon: Icons.edit_outlined,
          title: 'Manual Reminders',
          description: 'Create and send reminders manually as needed.',
          isSelected: _mainSelection == 'Manual Reminders',
          onTap: _selectManual,
        ),
      ],
    );
  }

  Widget _autoSubStep() {
    return Column(
      children: [
        _optionCard(
          icon: Icons.notifications_active,
          title: 'Overdue Payment Reminder',
          description: 'Send automatic reminders for payments that are past due date.',
          isSelected: _autoSubSelected == 0,
          onTap: () => setState(() => _autoSubSelected = 0),
          showBadge: true,
        ),
        const SizedBox(height: 8),
        _optionCard(
          icon: Icons.schedule,
          title: 'Upcoming Payment Reminder',
          description: 'Notify customers before payment due dates.',
          isSelected: _autoSubSelected == 1,
          onTap: () => setState(() => _autoSubSelected = 1),
        ),
      ],
    );
  }

  Widget _optionCard({
    required IconData icon,
    required String title,
    required String description,
    required bool isSelected,
    required VoidCallback onTap,
    bool showBadge = false,
  }) {
    return GestureDetector(
      onTap: onTap,
      child: Container(
        width: double.infinity,
        padding: const EdgeInsets.all(16),
        decoration: BoxDecoration(
          color: isSelected ? Colors.blue[50] : Colors.grey[100],
          borderRadius: BorderRadius.circular(12),
          border: Border.all(
            color: isSelected ? Colors.blue : Colors.transparent,
            width: 2,
          ),
        ),
        child: Row(
          children: [
            Icon(icon, color: isSelected ? Colors.blue : Colors.grey),
            const SizedBox(width: 12),
            Expanded(
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  Row(
                    children: [
                      Text(title, style: const TextStyle(fontWeight: FontWeight.w600)),
                      if (showBadge) ...[
                        const SizedBox(width: 8),
                        Container(
                          padding: const EdgeInsets.symmetric(horizontal: 6, vertical: 2),
                          decoration: BoxDecoration(
                            color: Colors.green,
                            borderRadius: BorderRadius.circular(4),
                          ),
                          child: const Text('Recommended',
                              style: TextStyle(color: Colors.white, fontSize: 9, fontWeight: FontWeight.w500)),
                        ),
                      ],
                    ],
                  ),
                  const SizedBox(height: 4),
                  Text(description, style: TextStyle(fontSize: 12, color: Colors.grey[600])),
                ],
              ),
            ),
          ],
        ),
      ),
    );
  }
}

File: app/src/main/java/com/bmstally/app/ui/screens/ReminderBottomSheet.kt
package com.bmstally.app.ui.screens

import androidx.compose.animation.AnimatedContent
import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

enum class ReminderStep { MAIN, AUTO_SUB }

@Composable
fun ReminderBottomSheet(
    onContinue: (String) -> Unit
) {
    var step by remember { mutableStateOf(ReminderStep.MAIN) }
    var mainSelection by remember { mutableStateOf<String?>(null) }
    var autoSubSelected by remember { mutableStateOf<Int?>(null) }

    val canContinue = when (mainSelection) {
        "Auto Reminders" -> autoSubSelected != null
        "Manual Reminders" -> true
        else -> false
    }

    Column(
        modifier = Modifier.padding(start = 16.dp, end = 16.dp, top = 16.dp, bottom = 16.dp)
    ) {
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically
        ) {
            if (step == ReminderStep.AUTO_SUB) {
                IconButton(onClick = { step = ReminderStep.MAIN; autoSubSelected = null }) {
                    Icon(Icons.Default.ArrowBack, "Back")
                }
            }
            Text(
                if (step == ReminderStep.MAIN) "Select Type of Reminder" else "Auto Reminders",
                fontWeight = FontWeight.Bold,
                fontSize = 18.sp,
                modifier = Modifier.weight(1f)
            )
        }

        Spacer(Modifier.height(12.dp))

        AnimatedContent(step) {
            when (it) {
                ReminderStep.MAIN -> MainStep(
                    mainSelection = mainSelection,
                    onAutoSelect = { mainSelection = "Auto Reminders"; step = ReminderStep.AUTO_SUB },
                    onManualSelect = { mainSelection = "Manual Reminders" }
                )
                ReminderStep.AUTO_SUB -> AutoSubStep(autoSubSelected) { autoSubSelected = it }
            }
        }

        Spacer(Modifier.height(8.dp))
        Text(
            "Reminders will be sent via email and SMS based on your configured schedule.",
            fontSize = 12.sp,
            color = Color.Gray
        )

        Spacer(Modifier.height(16.dp))
        Button(
            onClick = {
                if (mainSelection == "Manual Reminders") onContinue("manual")
                else onContinue("auto")
            },
            enabled = canContinue,
            modifier = Modifier.fillMaxWidth().height(48.dp)
        ) {
            Text("CONTINUE")
        }
    }
}

@Composable
private fun MainStep(mainSelection: String?, onAutoSelect: () -> Unit, onManualSelect: () -> Unit) {
    Column {
        OptionCard(
            icon = Icons.Default.Autorenew,
            title = "Auto Reminders",
            description = "Automatically send reminders based on predefined rules and schedules.",
            isSelected = mainSelection == "Auto Reminders",
            showBadge = true,
            onClick = onAutoSelect
        )
        Spacer(Modifier.height(8.dp))
        OptionCard(
            icon = Icons.Default.Edit,
            title = "Manual Reminders",
            description = "Create and send reminders manually as needed.",
            isSelected = mainSelection == "Manual Reminders",
            onClick = onManualSelect
        )
    }
}

@Composable
private fun AutoSubStep(selected: Int?, onSelect: (Int) -> Unit) {
    Column {
        OptionCard(
            icon = Icons.Default.NotificationsActive,
            title = "Overdue Payment Reminder",
            description = "Send automatic reminders for payments that are past due date.",
            isSelected = selected == 0,
            showBadge = true,
            onClick = { onSelect(0) }
        )
        Spacer(Modifier.height(8.dp))
        OptionCard(
            icon = Icons.Default.Schedule,
            title = "Upcoming Payment Reminder",
            description = "Notify customers before payment due dates.",
            isSelected = selected == 1,
            onClick = { onSelect(1) }
        )
    }
}

@Composable
private fun OptionCard(
    icon: androidx.compose.ui.graphics.vector.ImageVector,
    title: String,
    description: String,
    isSelected: Boolean,
    showBadge: Boolean = false,
    onClick: () -> Unit
) {
    Surface(
        modifier = Modifier.fillMaxWidth().clickable(onClick = onClick),
        shape = MaterialTheme.shapes.medium,
        color = if (isSelected) Color(0xFFE3F2FD) else Color(0xFFF5F5F5),
        border = if (isSelected) BorderStroke(2.dp, Color(0xFF3F51B5)) else null
    ) {
        Row(Modifier.padding(16.dp), verticalAlignment = Alignment.Top) {
            Icon(icon, null, tint = if (isSelected) Color(0xFF3F51B5) else Color.Gray)
            Spacer(Modifier.width(12.dp))
            Column(Modifier.weight(1f)) {
                Row(verticalAlignment = Alignment.CenterVertically) {
                    Text(title, fontWeight = FontWeight.W600)
                    if (showBadge) {
                        Spacer(Modifier.width(8.dp))
                        Surface(shape = MaterialTheme.shapes.small, color = Color(0xFF4CAF50)) {
                            Text(
                                "Recommended",
                                Modifier.padding(horizontal = 6.dp, vertical = 2.dp),
                                color = Color.White,
                                fontSize = 9.sp,
                                fontWeight = FontWeight.W500
                            )
                        }
                    }
                }
                Spacer(Modifier.height(4.dp))
                Text(description, fontSize = 12.sp, color = Color.Gray)
            }
        }
    }
}